### ISSUE TITLE
ARG003 and ARG004 don't get triggered

### ISSUE THREAD

**Issue by spapanik** [charliermarsh/ruff#4526]
Hi,

I ARG003 and ARG004 aren't working as expected.

My minimal example is a file containing:
```python
class Example:
    def can_be_static(self, x: int) -> int:
        return x

    @classmethod
    def cls_can_be_static(cls, x: int) -> int:
        return x
```

and the `pyproject.toml` contains (ruff-related) only:
```toml
[tool.ruff]
select = ["ARG"]
```

trying to lint the file using ruff produces no errors, although it was expected find both ARG003 and ARG004

my version is `ruff 0.0.269`

**Comment by JonathanPlasse** [comment]
> ```python
> class Example:
>     def can_be_static(self, x: int) -> int:
>         return x
>                ^
> 
>     @classmethod
>     def cls_can_be_static(cls, x: int) -> int:
>         return x
>                ^
> ```
I added `^` where `x` is used. `ARG` will ignore `self` and `cls`.

**Comment by charliermarsh** [comment]
Yeah this is working as intended I believe.

**Comment by spapanik** [comment]
Maybe I'm misunderstanding, because if in the first method I `return self`, then ARG002 would get triggered. I was under the impression that ARG004 would trigger if `self` is unused, as this means that the method can be turned into a static method.

If this isn't the case, I would like to make a feature request for a rule that checks for unused `self`, or unused `cls` in the case of classmethods.
